mesonbuild/backend/ninjabackend.py

```diff
@@ -2164,7 +2164,17 @@ def determine_swift_external_dep_link_args(self, target, swiftc):
             if isinstance(d, build.StaticLibrary):
                 for dep in d.get_external_deps():
                     args += swiftc.get_dependency_link_args(dep)
-        return args
+
+        deduped_args = []
+        seen_libs = set()
+        for arg in args:
+            if arg.startswith("-l"):
+                if arg not in seen_libs:
+                    deduped_args.append(arg)
+                    seen_libs.add(arg)
+            else:
+                deduped_args.append(arg)
+        return deduped_args
```

Comment:
This seems to be the sort of thing that `CompileArgs` exists for, is it possible to use that for this purose?